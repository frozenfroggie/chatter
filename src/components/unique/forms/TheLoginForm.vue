<template>
  <div class="inputContainer">
    <div class="username">
      <icon name="user" scale="1"></icon>
      <input
        type="text"
        name="username"
        placeholder="Username"
        aria-label="Username"
        v-model="username">
    </div>
    <div class="password">
      <icon name="unlock" scale="1"></icon>
      <input
        type="password"
        name="password"
        placeholder="Password"
        aria-label="Password"
        v-model="password">
    </div>
  </div>
</template>

<script>

export default {
  data () {
    return {
      username: '',
      password: ''
    }
  }
}
</script>

<style scoped>
.inputContainer {
  grid-area: loginInput;
  background-color: rgb(38, 49, 66);
}
.username {
  height: 50%;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.password {
  height: 50%;
  border-top: 1px solid rgba(250,250,250,0.5);
  border-bottom: 1px solid rgba(250,250,250,0.5);
}
input {
  height: 100%;
  width: 80%;
  padding-left: 10px;
  background-color: inherit;
  border: none;
  color: rgba(250,250,250,0.9);
}
@media only screen and (min-width: 768px) {
  .inputContainer {
    border-radius: 10px;
  }
  .username {
    border-radius: 10px 10px 0px 0px;
  }
  .password {
    border-radius: 0px 0px 10px 10px;
  }
}
</style>
