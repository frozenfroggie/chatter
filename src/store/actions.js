export default {
  updateMessage: ({commit}, payload) => {
    console.log('update!!!', payload)
    commit('updateMessage', payload)
  },
  addEmojiMessage: ({commit}, payload) => {
    console.log('update emoji!!!', payload)
    commit('addEmojiMessage', payload)
  },
  removeMessage: ({commit}) => {
    commit('removeMessage')
  },
  setSendMessageIntention: ({commit}, payload) => {
    commit('setSendMessageIntention', payload)
  },
  typingNotification: ({commit}, payload) => {
    commit('toogleIsTyping')
  },
  openEmojiPanel: ({commit}, payload) => {
    commit('setEmojiPanel', true)
  },
  closeEmojiPanel: ({commit}, payload) => {
    commit('setEmojiPanel', false)
  },
  showVideoChatPrompt: ({commit}) => {
    commit('showVideoChatPrompt')
  },
  videoChatStart: ({commit}) => {
    commit('videoChatStart')
  },
  videoChatHangup: ({commit}) => {
    commit('videoChatHangup')
  }
}
