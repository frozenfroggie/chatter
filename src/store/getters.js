export default {
  messageText: state => state.messageText,
  addEmoji: state => state.addEmoji,
  emojiPanel: state => state.emojiPanel,
  sendMessageIntention: state => state.sendMessage.intention,
  isTyping: state => state.isTyping,
  videoChatCalling: state => state.videoChat.isCalling,
  isVideoChatStarted: state => state.videoChat.isStarted,
  isVideoChatHangup: state => state.videoChat.hangup
}
