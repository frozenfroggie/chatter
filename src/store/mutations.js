export default {
  emailSend: (state) => {
    state.tokens.verificationToken.send = true
  },
  updateMessage: (state, payload) => {
    state.messageText = payload
  },
  addEmojiMessage: (state, payload) => {
    state.addEmoji = payload
    state.messageText += payload
  },
  removeMessage: (state) => {
    state.messageSend = true
  },
  sendMessagePending: (state, payload) => {
    state.sendMessage.pending = true
  },
  sendMessageSuccess: (state, payload) => {
    state.sendMessage.pending = false
    state.sendMessage.success = true
  },
  setSendMessageIntention: (state, payload) => {
    state.sendMessage.intention = payload
  },
  toogleIsTyping: (state) => {
    state.isTyping = !state.isTyping
  },
  setEmojiPanel: (state, payload) => {
    state.emojiPanel = payload
  },
  showVideoChatPrompt: (state) => {
    state.videoChat.isCalling = true
    state.videoChat.hangup = false
  },
  videoChatStart: (state) => {
    state.videoChat.isCalling = false
    state.videoChat.isStarted = true
    state.videoChat.hangup = false
  },
  videoChatHangup: (state) => {
    state.videoChat.isCalling = false
    state.videoChat.isStarted = false
    state.videoChat.hangup = true
  }
}
