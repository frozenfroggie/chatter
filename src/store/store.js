import Vue from 'vue'
import Vuex from 'vuex'
import VuexPersistence from 'vuex-persist'

import forms from './modules/forms'
import conversations from './modules/conversations'
import getMessages from './modules/getMessages'
import searchUsers from './modules/searchUsers'
import socket from './modules/socket'
import user from './modules/user'
import resizer from './modules/resizer'

import mutations from './mutations'
import getters from './getters'
import actions from './actions'

Vue.use(Vuex)

const vuexLocal = new VuexPersistence({
  storage: window.localStorage,
  supportCircular: true,
  modules: ['user', 'tokens', 'conversations']
})

export const store = new Vuex.Store({
  state: {
    socket: null,
    tokens: {
      accessToken: null,
      refreshToken: null,
      verificationToken: {
        send: false,
        confirmed: false
      }
    },
    messageText: '',
    addEmoji: '',
    emojiPanel: false,
    sendMessage: {
      intention: false,
      pending: false,
      success: false,
      error: false
    },
    isTyping: false,
    videoChat: {
      isCalling: false,
      isStarted: false,
      hangup: false
    }
  },
  getters,
  mutations,
  actions,
  modules: {
    forms,
    conversations,
    getMessages,
    searchUsers,
    socket,
    user,
    resizer
  },
  plugins: [vuexLocal.plugin]
})
