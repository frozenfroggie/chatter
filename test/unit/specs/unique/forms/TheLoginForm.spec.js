import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import TheLoginForm from '@/components/unique/forms/TheLoginForm'
import { store } from '@/store/store'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('TheLoginForm', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(TheLoginForm, {
      stubs: ['icon'],
      localVue,
      store
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
