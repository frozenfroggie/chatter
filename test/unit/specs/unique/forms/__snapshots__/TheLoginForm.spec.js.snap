// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheLoginForm matches snapshot 1`] = `
<div class="inputContainer">
    <div class="username">
        <!---->
        <input type="text" name="username" placeholder="Username" aria-label="Username">
    </div>
    <div class="password">
        <!---->
        <input type="password" name="password" placeholder="Password" aria-label="Password">
    </div>
</div>
`;
